waku/v2/protocol/waku_filter/waku_filter.nim

```diff
@@ -67,11 +66,12 @@ proc unsubscribeFilters(subscribers: var seq[Subscriber], request: FilterRequest
     # Iterate through subscriber entries matching peer ID to remove matching content topics
     for cf in subscriber.filter.contentFilters.mitems:
       # Iterate content filters in filter entry
-      cf.contentTopics.keepIf(proc (t: auto): bool = t notin unsubscribeTopics)
+      if cf.contentTopic in unsubscribeTopics:
+        cf.contentTopic = ""
 
     # make sure we delete the content filter
     # if no more topics are left
-    subscriber.filter.contentFilters.keepIf(proc (cf: auto): bool = cf.contentTopics.len > 0)
+    subscriber.filter.contentFilters.keepIf(proc (cf: auto): bool = cf.contentTopic.len > 0)
```

Comment:
We had to do two passes previously:
1 to remove individual, unsubscribing `contentTopic`s from the nested sequences
1 to ensure that `contentFilter`s with no more `contentTopic`s in their nested sequences are removed

Now that the `contentFilter` only has a single `contentTopic`, you can likely replace this whole section now with only a single pass, something like
```
subscriber.filter.contentFilters.keepIf(proc (cf: auto): bool = cf.contentTopic notin unsubscribeTopics
```